apps/server/src/migrations/mikro-orm/Migration20250912083440.ts

```diff
@@ -0,0 +1,369 @@
+import { Migration } from '@mikro-orm/migrations-mongodb';
+import { FindCursor, WithId } from '@mikro-orm/mongodb/node_modules/mongodb';
+import { AesEncryptionHelper } from '@shared/common/utils';
+import CryptoJS from 'crypto-js';
+
+const storageProvidersCollectionName = 'storageproviders';
+const systemsCollectionName = 'systems';
+const externalToolsCollectionName = 'external-tools';
+const mediaSourcesCollectionName = 'media-sources';
+
+// Update all AES encrypted secrets with new encryption function.
+export class Migration20250710083440 extends Migration {
+	public async up(): Promise<void> {
+		//  --- Update secretAccessKey of storage providers ---
+		// eslint-disable-next-line no-process-env
+		const { S3_KEY } = process.env;
+
+		if (!S3_KEY) {
+			console.error('S3_KEY is not provided. Migration cannot proceed.');
+			return;
+		}
+
+		const storageProviders = this.getCollection(storageProvidersCollectionName).find({});
+
+		const numberOfUpdatedStorageProviders = await this.updateSecrets(
+			storageProviders,
+			['secretAccessKey'],
+			S3_KEY,
+			storageProvidersCollectionName
+		);
+
+		console.info(
+			`Updated secretAccessKey of ${numberOfUpdatedStorageProviders} storage providers with new encryption function.`
+		);
+
+		// ----------------------------------------------------------------------------------
+
+		// --- Update searchUserPassword of LDAP systems ---
+		// eslint-disable-next-line no-process-env
+		const { LDAP_PASSWORD_ENCRYPTION_KEY } = process.env;
+
+		if (!LDAP_PASSWORD_ENCRYPTION_KEY) {
```

Comment:
if we want to stop the whole migration, in case one of the envs is not found, then have all the verification on top before any database queries. 
This way we don't end up in a state when some collections were updated and others not. The rollback of the partially updated collections, could also be more messy in this case.

Reply:
Yes, I think that's better. I changed it.